util/embedded-ldap/src/main/java/org/keycloak/util/ldap/LDAPEmbeddedServer.java

```diff
@@ -226,19 +251,52 @@ protected LdapServer createLdapServer() {
         // Read the transports
         Transport ldap = new TcpTransport(this.bindHost, this.bindPort, 3, 50);
         ldapServer.addTransports( ldap );
-        if (enableSSL) {
-            Transport ldaps = new TcpTransport(this.bindHost, this.bindLdapsPort, 3, 50);
-            ldaps.setEnableSSL(true);
+        if (enableSSL || enableStartTLS) {
             ldapServer.setKeystoreFile(keystoreFile);
             ldapServer.setCertificatePassword(certPassword);
-            ldapServer.addTransports( ldaps );
+            if (enableSSL) {
+                Transport ldaps = new TcpTransport(this.bindHost, this.bindLdapsPort, 3, 50);
+                ldaps.setEnableSSL(true);
+                ldapServer.addTransports( ldaps );
+                if (ldaps.isSSLEnabled()) {
+                    log.info("Enabled SSL support on the LDAP server.");
+                }
+            }
+            if (enableStartTLS) {
+                try {
+                    ldapServer.addExtendedOperationHandler(new StartTlsHandler());
+                } catch (Exception e) {
+                    throw new IllegalStateException("Cannot add the StartTLS extension handler: ", e);
+                }
+                for (ExtendedOperationHandler eoh : ldapServer.getExtendedOperationHandlers()) {
+                    if (eoh.getOid().equals(StartTlsHandler.EXTENSION_OID)) {
+                        log.info("Enabled StartTLS support on the LDAP server.");
+                        break;
+                    }
+                }
+            }
         }
 
         // Associate the DS to this LdapServer
         ldapServer.setDirectoryService( directoryService );
 
-        // Propagate the anonymous flag to the DS
-        directoryService.setAllowAnonymousAccess(false);
+        if (enableAnonymousAccess && !enableAccessControl) {
```

Comment:
AFAIU it's not completely forbidden combination (per the Apache DS upstream docs), but once the access control subsystem is enabled, "_full_" anonymous binds / access might not work even if anonymous access is enabled (only root DSE would be allowed to be accessed directly, the child ones not, here the access control setting would apply).

But agree it will be better not to allow this combination (both access control & anonymous binds enabled @ the same time) to simplify the configuration for the test runs. This way anonymous binds are either fully enabled or disabled. There's nothing in between (the test writer won't need to find out / solve the nuisances between full & partial anonymous access allowed).

Fixed in 2nd commit